### PR TITLE
A new, modern, fresh look for the Scratch site! #7722

### PR DESCRIPTION
**Pull request by blazgocompany**
A new, modern, fresh look for the Scratch site • #7722

### Changes

Changed the styles for the profile page and the home page. More styles for more pages will come soon

### Reason for changes

Makes the Scratch site look better.

### Tests

Looks good on Firefox and Brave:
image

I don't know if any other addons are interfering with my code.

### CODE REVIEW

File: addons/modern-home-page/addon.json [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    "dynamicDisable": true,
+    "versionAdded": "1.38.5",
+    "userstyles": [
+      {
+        "url": "home.css",
+        "matches": ["https://scratch.mit.edu"]
+      },
+      {
+        "url": "profile.css",
+        "matches": ["users"]
```
Comment (Samq64) [inline addons/modern-home-page/addon.json:20]:
```suggestion
        "matches": ["profiles"]
```

File: addons/modern-home-page/home.css [hunk 2]
```diff
@@ -0,0 +1,102 @@
+:root {
+    --ste-blue: #4e97fe;
```
Comment (Samq64) [inline addons/modern-home-page/home.css:2]: Is this a reference to ScratchTools? If so, could you provide a link to the original version?

**Review by Samq64 — CHANGES_REQUESTED** [review]
Without any other addons enabled, hovering over what's happening items causes the others to shift and there are multiple issues with profiles. Currently this is also incompatible with dark mode and Scratch 2.0 → 3.0.

File: addons/modern-home-page/home.css (hunk 2 above)
Comment (blazgocompany) [reply addons/modern-home-page/home.css:2]: Yes, this is my own addon from ScratchTools which has not been published yet. (I can give the PR link)

**Comment by blazgocompany** [comment]
> Currently this is also incompatible with dark mode and Scratch 2.0 → 3.0

Why is it incompatible with dark mode? Do you have a screen shot? Are you talking about the dark mode addon or dark mode preferences in the browser?

Also, this has nothing to do with Scratch 3.0. These are enhanced styles for the **entire** Scratch site, **regardless** of whether they are currently 2.0 or 3.0-styled pages. Consider this as changing **everything** to Scratch "4.0"-styled pages.

**Comment by blazgocompany** [comment]
> @Samq64
> Currently this is also incompatible with dark mode and Scratch 2.0 → 3.0


Why is it incompatible with dark mode? Do you have a screen shot? Are you talking about the dark mode addon or dark mode preferences in the browser?

Also, this has nothing to do with Scratch 3.0. These are enhanced styles for the **entire** Scratch site, **regardless** of whether they are currently 2.0 or 3.0-styled pages. Consider this as changing **everything** to Scratch "4.0"-styled pages.

**Comment by blazgocompany** [comment]
> @Samq64 
> there are multiple issues with profiles

Can you please provide a screenshot of the issues in the profile page?

> @Samq64 
> hovering over what's happening items causes the others to shift

I'll try to fix that, but I don't know if it's possible.

File: addons/modern-home-page/home.css [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    border-radius: 0px 0px 20px 20px;
+  }
+
+  .box-header > h4{
+    color: black !important;
+  }
+  .box-header > a > p{
+    color: black !important
+  }
+  .carousel .slick-slide.project{
```
Comment (blazgocompany) [inline addons/modern-home-page/home.css:64]: This should hopefully fix the issue with the "What's Happening" items shifting

**Comment by blazgocompany** [comment]
@Samq64, I've fixed the issue where the "What's Happening" items shift.

**Comment by Samq64** [comment]
> Also, this has nothing to do with Scratch 3.0. These are enhanced styles for the entire Scratch site, regardless of whether they are currently 2.0 or 3.0-styled pages.

We want to make sure nothing looks broken regardless of what combination of addons users have enabled. With both enabled there are a couple of issues like a cut off profile picture and incorrect text colour without dark mode in some places.

The homepage with dark mode:
image
Profiles with dark mode:
image
And Scratch 2.0 → 3.0:
image

**Comment by Samq64** [comment]
For dark mode support usually addons do something like `var(--DarkWww-variableName, #abc123)`

**Comment by blazgocompany** [comment]
The home page with dark mode certainly looks odd. I'll try to fix that.
 I don't quite understand your images for the second and third ones. Are they edited (screenshot of light mode pasted of screenshot of dark)?

**Comment by Samq64** [comment]
> I don't quite understand your images for the second and third ones. Are they edited (screenshot of light mode pasted of screenshot of dark)?

No it's just a solid background that seems transparent in light mode.

I also thought the extra padding under the profile picture was was bug.

**Comment by blazgocompany** [comment]
Ok, back.

I would appreciate your thoughts on the profile picture actually. I can't make it too big because not everyone has high-quality profile pictures.

**Comment by blazgocompany** [comment]
@Samq64 Please let me know if there are any other bugs you may find.

**Comment by blazgocompany** [comment]
@Samq64 Please let me know if there are any other bugs you may find.

**Comment by WorldLanguages** [comment]
The associated issue was created 10 minutes before submitting this PR, defeating its purpose.

**Comment by WorldLanguages** [comment]
I haven't looked at this deeply, but this looks quite opinionated (who defines what is modern?), and the scope of the addon could grow infinitely (dozens of pages to support). It will also conflict with other themes. I find it unlikely this will ever get the necessary approvals to get into the extension.

**Comment by blazgocompany** [comment]
@WorldLanguages, I understand your point there, but one part of the goal is to make the Scratch site **consistent**.
Of course, right now, with half styled and half unstyled, things can look pretty inconsistent, which is why this is a beta addon. As we continue to develop this addon, the Scratch site will become more and more consistent.

> Who defines what is modern?

Well, who decides that the current Scratch website is "good"? Not everyone likes it, but Scratch did it anyways. Also, these styles are not just my suggestions, some of these were [actually designed and were ❤️+⭐ by over 5000 people](https://scratch.mit.edu/projects/827943393/)

**Comment by blazgocompany** [comment]
Let me know if this addon meets the criteria to be added.

**Comment by mxmou** [comment]
> @WorldLanguages, I understand your point there, but one part of the goal is to make the Scratch site **consistent**.

I don't think redesigning everything is the best way to make it consistent. In https://github.com/ScratchAddons/ScratchAddons/issues/3470#issuecomment-917645557, I suggested a design for the front page that would make it more similar to other Scratch 3.0-styled pages, which could be implemented as an option in the Scratch 2.0 → 3.0 addon.

> Also, these styles are not just my suggestions, some of these were [actually designed and were ❤️+⭐ by over 5000 people.](https://scratch.mit.edu/projects/827943393/)

There are many "Scratch 4.0" proposals and it would be difficult to add all of them to Scratch Addons because all new themes have to be compatible with existing ones (website dark mode, old studio layout...). Instead, we can add more customization options so that users can create their preferred design. For this specific suggestion: making the background of boxes blue is already possible, removing the border should be possible but currently doesn't work (I'll open a PR to fix that) and making them entirely rounded could be an option in 2.0 → 3.0.

There's no option to add a hover background to What's Happening items. Only clickable elements should have a hover state.

**Comment by blazgocompany** [comment]
> There's no option to add a hover background to What's Happening items.

I think you meant no "reason". I'll remove the hover states.

> For this specific suggestion: making the background of boxes blue is already possible, removing the border should be possible but currently doesn't work

Border of what? I think I have removed the border from all `.box` containers. A screenshot of where those borders are would help.

> Because all new themes have to be compatible with existing ones (website dark mode, old studio layout...)

Most of the code here will not be affected by other addons. Even if they did, people could turn off those and just customize this current addon.

@WorldLanguages, I didn't quite understand what you **want** me to do. Do you want me to change an existing addon? Do you want me to add customizations to this addon? Do you want to to just forget about it?

**Comment by GrahamSH-LLK** [comment]
> @WorldLanguages, I understand your point there, but one part of the goal is to make the Scratch site **consistent**. Of course, right now, with half styled and half unstyled, things can look pretty inconsistent, which is why this is a beta addon. As we continue to develop this addon, the Scratch site will look more and more consistent.
> 
> > Who defines what is modern?
> 
> Well, who decides that the current Scratch website is "good"? Not everyone likes it, but Scratch did it anyways. Also, these styles are not just my suggestions, some of these were [actually designed and were ❤️+⭐ by over 5000 people.](https://scratch.mit.edu/projects/827943393/)
> 
> Let me know if this addon meets the criteria to be added.

This makes the site less consistent. We already have addons like scratchr2 for that. I do not think this addon is in scope for this project. You said you already PRd to STE- we also don’t want to be duplicating their features. Strong no support.

**Comment by blazgocompany** [comment]
Ok, I'll close this then

**Comment by GrahamSH-LLK** [comment]
> > You said you already PRd to STE- we also probably don’t want to be duplicating their features
> 
> I don't know who's "we". But it is nice that contributors have alternative projects to submit their changes to.

that's why it had a question mark after it, I wasn't sure

**Pull request closed without merging**